packages/react/radio-group/src/RadioGroup.tsx

```diff
@@ -103,58 +109,89 @@ RadioGroup.displayName = RADIO_GROUP_NAME;
 
 const ITEM_NAME = 'RadioGroupItem';
 
-type RadioGroupItemOwnProps = Merge<
+type RadioGroupItemOwnProps = Polymorphic.OwnProps<typeof RadioGroupItemImpl>;
+type RadioGroupItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof RadioGroupItemImpl>,
+  RadioGroupItemOwnProps
+>;
+
+const RadioGroupItem = React.forwardRef((props, forwardedRef) => {
+  const context = useRadioGroupContext(ITEM_NAME);
+  return context.rovingFocus ? (
+    <RadioGroupRovingFocusItem {...props} ref={forwardedRef} />
+  ) : (
+    <RadioGroupItemImpl {...props} ref={forwardedRef} />
+  );
+}) as RadioGroupItemPrimitive;
+
+type RadioGroupRovingFocusItemOwnProps = Polymorphic.OwnProps<typeof RadioGroupItemImpl>;
+type RadioGroupRovingFocusItemPrimitive = Polymorphic.ForwardRefComponent<
+  Polymorphic.IntrinsicElement<typeof RadioGroupItemImpl>,
+  RadioGroupRovingFocusItemOwnProps
+>;
+
+const RadioGroupRovingFocusItem = React.forwardRef((props, forwardedRef) => {
```

Comment:
I like this new separation now, and seeing the event handlers only for the roving focus one 👍